packages/Tailwind/Ignis.Components.HeadlessUI/Aria/AriaPopupExtensions.cs

```diff
@@ -41,23 +43,34 @@ public static void OnKeyDown(this IAriaPopup popup, KeyboardEventArgs eventArgs)
                 break;
             case "ArrowUp" when popup.ActiveDescendant == null:
             case "ArrowDown" when popup.ActiveDescendant == null:
-                if (descendants.Any()) popup.SetActiveDescendant(descendants[0], isActive: true);
+                if (descendants.Any()) popup.SetActiveDescendant(descendants[0], isActive: true, jSRuntime);
                 else if (!popup.IsOpen) popup.Open();
                 break;
             case "ArrowDown":
                 {
                     var index = Array.IndexOf(descendants, popup.ActiveDescendant) + 1;
-                    if (index < descendants.Length) popup.SetActiveDescendant(descendants[index], isActive: true);
+                    if (index < descendants.Length) popup.SetActiveDescendant(descendants[index], isActive: true, jSRuntime);
                     else if (!popup.IsOpen) popup.Open();
                     break;
                 }
             case "ArrowUp":
                 {
                     var index = Array.IndexOf(descendants, popup.ActiveDescendant) - 1;
-                    if (index >= 0) popup.SetActiveDescendant(descendants[index], isActive: true);
+                    if (index >= 0) popup.SetActiveDescendant(descendants[index], isActive: true, jSRuntime);
                     else if (!popup.IsOpen) popup.Open();
                     break;
                 }
         }
     }
+
+    private static object ScrollOptions { get; } = new { Behavior = "smooth", Block = "nearest" };
+
+    private static async Task ScrollIntoView(string? id, IJSRuntime? jSRuntime)
```

Comment:
I refrained from doing it as to not change the `Inetrace` as it would be a breaking change. If I change it should I also pass an `scroll` flag so that the code is not always run?

Reply:
What would be the breaking change here? Also why not run it always since it should be run on all popups anyways?

Reply:
Breaking as in adding a new property to an existing interface but it makes sense to me too and as you are onboard I will be adding it 😄 .

For not calling it always I thought of the `ArrowDown` + `ActiveDescendant == null` scenario where there is no need to scroll since the descendand is the first and the popup will already be at the top. For this only the server comes to mind as doing too much but it could be negligible.

Reply:
True, I am proceeding with these changes then 😀